internal/pkg/deploy/cloudformation/app.go

```diff
@@ -50,7 +52,66 @@ func (cf CloudFormation) DeployApp(in *deploy.CreateAppInput) error {
 }
 
 func (cf CloudFormation) UpgradeApplication(in *deploy.CreateAppInput) error {
-	return nil
+	appConfig := stack.NewAppStackConfig(in)
+	s, err := toStack(appConfig)
+	if err != nil {
+		return err
+	}
+	if err := cf.upgradeAppStack(s); err != nil {
+		return err
+	}
+	return cf.upgradeAppStackSet(appConfig)
+}
+
+func (cf CloudFormation) upgradeAppStackSet(config *stack.AppStackConfig) error {
+	for {
+		ssName := config.StackSetName()
+		if err := cf.appStackSet.WaitForStackSetLastOperationComplete(ssName); err != nil {
```

Comment:
good question! stackset only allows one operation at a time otherwise it will throw
```
//   * ErrCodeOperationInProgressException "OperationInProgressException"
//   Another operation is currently in progress for this stack set. Only one operation
//   can be performed for a stack set at a given time.
```
We don't handle updating stackset instances directly. We just update the stackset and by default it will automatically populate changes to all associated accounts/regions *sequentially* (previously, now we can switch to parallel).

Reply:
Thanks! Do we need to handle that exception then and then retry the upgrade operation?

Reply:
It is actually already handled! https://github.com/aws/copilot-cli/blob/dea26599f8ec029c3d205bc72f4ee4ab36bc1847/internal/pkg/aws/cloudformation/stackset/errors.go#L39